tests/test_api.py

```diff
@@ -69,6 +69,60 @@ def test_get_version(self):
         res = self.assertGetResponse("/version", requests.codes.ok, headers=get_auth_header())
         self.assertEquals(res.json['version_info']['version'], os.environ['DSS_VERSION'])
 
+    @testmode.standalone
+    def test_put_file_pattern(self):
+        """
+        Tests the regex pattern filtering on the PUT file/{uuid} endpoint.
+
+        Ensures that paths with leading slashes (absolute paths) and
+        relative path shortcuts (".", "~/", and "..") are disallowed.
+        """
+
+        examples_of_bad_paths = ['.', '..',
+                                 '~/path',
+                                 './', '../', './path', './../path', '../path', '../../path',
+                                 'path/../path', 'path/..',
+                                 '/path', '/path.json', '/bad..path']
+
+        examples_of_good_paths = ['path', 'path.json', 'good..path', 'path.json/', 'good..path/',
+                                  'pa/th.json', 'go/od..path', 'a/b/c/d', 'a/b.c/d', '.bashrc']
+
+        for path, replica in [("files", "aws"), ("files", "gcp")]:
+            with self.subTest(path=path, replica=replica):
```

Comment:
The subtest should be something like this
``` python
        for bad_path in examples_of_bad_paths:
            with self.subTest(path=bad_path, replica=replica):
                    self.put_file_reponse(bad_path, replica, expected_code=requests.codes.bad_request)
```

Do the samething for good_path

Reply:
Why?

Reply:
Thanks for the catch.  Looking back, I don't know why I did it that way.